pkg/api/types.go

```diff
@@ -1410,6 +1412,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
```

Comment:
> It's going to be tedious to keep a backpointer up to date.
> 
> Examples:
> 
> Remove or change a label to orphan a pod for debugging.
> Delete a controller and orphan its pods.
> Adopt a pod under a new controller.
> Create a new service for a set of pods.

I understand these reasons, but using a selector has its own problems.  I would rather avoid defining semantics for multiple DBs applying to the same pod, and implementation becomes more complicated, too.  A pod *could* point to an arbitrary number of DBs, but without transactions or some other form of serialization, we can't guarantee that we've gotten approval from all the DBs and also charged it against all of them.